docs/src/main/sphinx/language/lexical-structure.rst

```diff
@@ -0,0 +1,135 @@
+=================
+Lexical structure
+=================
+
+A SQL statement consist of a series of tokens, separated by whitespace
+characters, and terminated by a semicolon.
+
+A token can one be any of the following:
+
+* A keyword
+* An identifier
+* A quoted identifier
+* A literal (constant)
+* A special character symbol
+
+Valid whitespace characters are space, tab, or a newline.
+
+Statements can also contain comments, but they are not tokens, so they are
+effectively ignored.
+
+The following example contains all elements::
+
+    SELECT * FROM example_table WHERE a_column = '5'; -- a comment
+
+* ``SELECT``, ``FROM``, and ``WHERE`` are tokens.
+* The ``=`` operator, the  ``*`` symbol and the semicolon ``;`` are all  special
+  characters.
+* Space acts as whitespace to separate elements.
+* ``example_table``, and ``a_column`` are identifiers.
+* ``--`` starts a comment section for the rest of the line.
+
+Keywords and identifiers
+------------------------
+
+Keyword tokens, such as ``SELECT``, have a fixed meaning in the SQL language.
+Tokens that identify names of tables, columns, functions, or other objects, are
+identifiers. Keywords and identifiers have the same lexical structure, meaning
+that one cannot know whether a token is an identifier or a key word without
+knowing the language.
+
+The :doc:`list of reserved keywords </language/reserved>` includes all tokens
+used for the supported :doc:`/sql`.
+
+Identifiers must start with a letter and subsequently include letters, digits,
+and underscores. A dot is used to compose multiple identifiers to a of a
+qualified identifier. Identifiers with special characters must be quoted.
+Identifiers are not treated as case sensitive. Following are some valid
+examples:
+
+.. code-block:: text
+
+    tablename
+    TableName
+    example_catalog.a_schema."table$partitions"
+
+
+TODO: TIL there's support for backquoted identifiers, do we want to mention that too?
+TODO: unicode?
+
+Literals
+--------
+
+String literals
+^^^^^^^^^^^^^^^
+
+Single quotes, escaping them.
+Unicode Escapes
```

Comment:
We could start by copying this paragraph: https://trino.io/docs/current/language/types.html#varchar

Also, maybe it's worthing referencing this page in the data types section? A few times I was looking for those unicode escape codes in docs and couldn't remember where they are.